server/src/cores/storage.core.ts

```diff
@@ -115,7 +115,11 @@ export class StorageCore {
   }
 
   static isImmichPath(path: string) {
-    return resolve(path) === resolve(APP_MEDIA_LOCATION);
+    const normalizedPath = resolve(path).endsWith('/') ? resolve(path) : `${resolve(path)}/`;
```

Comment:
I'm also not sure if it's necessary to add a slash to that path anyways?

Reply:
At least the official docs https://nodejs.org/api/path.html#pathresolvepaths seem to imply this. Or are you aware of a scenario where that isn't the case?

Reply:
I think `/` is the exception where we wouldn't want to add a forward slash.

Reply:
> I'm also not sure if it's necessary to add a slash to that path anyways?

If we don't we run into the issue we're currently experiencing; `/foo-bar` starts with `/foo`, even though both are different folders and `/foo` is the media location

Reply:
The media location path needs a slash added, but the `path` that is passed in for checking doesn't.

Reply:
But we need to resolve the media location as that can also be a relative path, right? @bo0tzz

Reply:
Making sure both paths are in the same format before you compare them is a smart move.
What is the final decision?

Reply:
Probably yes. I think the best way forward would be to create some unit tests for this that spell out all the different paths.

Reply:
> Making sure both paths are in the same format before you compare them is a smart move. What is the final decision?

After all this discussion I think what you currently is the best (as in most resilient). However, I would like to hear the others chime in as well :)

Reply:
this is a long discussion about how it might or might not work, but at the end of the day we should _definitely_ just add a few unit tests for the situations we're worried about and make sure it is handling them correctly. The easiest one would just be `/photos_old` and `/photos`